feat: add clj-kondo optional profile (+kondo)

stevejmp opened the pull request:
Fixes #115

Followed directions for adding a new profile at [README: Contributing](https://github.com/day8/re-frame-template#contributing). Happy new year!

Tested as follows:

1. Create a kitchen sink, all-option project:
```sh
PROJDIR=`pwd`
DIFFDIR=`mktemp -d`

# Create a project with all options
lein install
lein new re-frame awesome-app +10x +breaking-point +cider +garden +handler +kondo +less +re-com +re-frisk +re-pressed +routes +test

```
2. Verify tests and builds run
```sh
cd awesome-app && npm install && lein deps
lein karma
lein dev
cd ..
```
3. Create a project for each option, moving READMEs and gitignores to the all-option project:
```sh
# Move all-option project to DIFFDIR
mv awesome-app $DIFFDIR

lein new re-frame awesome-app                 && mv awesome-app/README.md $DIFFDIR/awesome-app/README-0.md              && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-0 && rm -R awesome-app
lein new re-frame awesome-app +10x            && mv awesome-app/README.md $DIFFDIR/awesome-app/README-10x.md            && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-10x && rm -R awesome-app
lein new re-frame awesome-app +breaking-point && mv awesome-app/README.md $DIFFDIR/awesome-app/README-breaking-point.md && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-breaking-point && rm -R awesome-app
lein new re-frame awesome-app +cider          && mv awesome-app/README.md $DIFFDIR/awesome-app/README-cider.md          && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-cider && rm -R awesome-app
lein new re-frame awesome-app +garden         && mv awesome-app/README.md $DIFFDIR/awesome-app/README-garden.md         && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-garden && rm -R awesome-app
lein new re-frame awesome-app +handler        && mv awesome-app/README.md $DIFFDIR/awesome-app/README-handler.md        && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-handler && rm -R awesome-app
lein new re-frame awesome-app +kondo          && mv awesome-app/README.md $DIFFDIR/awesome-app/README-kondo.md          && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-kondo && rm -R awesome-app
lein new re-frame awesome-app +less           && mv awesome-app/README.md $DIFFDIR/awesome-app/README-less.md           && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-less && rm -R awesome-app
lein new re-frame awesome-app +re-com         && mv awesome-app/README.md $DIFFDIR/awesome-app/README-re-com.md         && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-re-com && rm -R awesome-app
lein new re-frame awesome-app +re-frisk       && mv awesome-app/README.md $DIFFDIR/awesome-app/README-re-frisk.md       && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-re-frisk && rm -R awesome-app
lein new re-frame awesome-app +re-pressed     && mv awesome-app/README.md $DIFFDIR/awesome-app/README-re-pressed.md     && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-re-pressed && rm -R awesome-app
lein new re-frame awesome-app +routes         && mv awesome-app/README.md $DIFFDIR/awesome-app/README-routes.md         && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-routes && rm -R awesome-app
lein new re-frame awesome-app +test           && mv awesome-app/README.md $DIFFDIR/awesome-app/README-test.md           && mv awesome-app/.gitignore $DIFFDIR/awesome-app/.gitignore-test && rm -R awesome-app
```
4. Upload the project to github and manually look over `README`s, checking links, etc.
5. Inspect `README` and `.gitignore` diffs:
```sh
# Inspect the differences between each option-project's README and the no-option-project README
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-10x.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-breaking-point.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-cider.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-garden.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-handler.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-kondo.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-less.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-re-com.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-re-frisk.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-re-pressed.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-routes.md
diff $DIFFDIR/awesome-app/README-0.md $DIFFDIR/awesome-app/README-test.md

# Inspect the differences between each option-project's gitignore and the no-option-project gitignore
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-10x
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-breaking-point
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-cider
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-garden
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-handler
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-kondo
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-less
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-re-com
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-re-frisk
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-re-pressed
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-routes
diff $DIFFDIR/awesome-app/.gitignore-0 $DIFFDIR/awesome-app/.gitignore-test
```
6. Clean up:
```sh
rm -rf $DIFFDIR
```

After removing unused requires, tested project variations using re-pressed with and without re-com, routes.

superstructor commented:
Thanks again @stevejmp and Happy new decade!

My understanding is that the re-pressed requires were removed as unused requires caused linter warnings ?

Also, I see you linked for emacs editor integration may be worth adding a link directly for other editors like Cursive and VS Code to https://github.com/borkdude/clj-kondo/blob/master/doc/editor-integration.md although that is also linked from the clj-kondo front-page.

stevejmp commented:
> My understanding is that the re-pressed requires were removed as unused requires caused linter warnings ?

Yes. The require is used in both [`views.cljs`](../../src/leiningen/new/re_frame/src/cljs/views.cljs) and [`views_recom.cljs`](../../src/leiningen/new/re_frame/src/cljs/views_recom.cljs), implementing an `::rp/set-keydown-rules` namespaced keyword.

If the `+routes` optional profile is used, however, the re-pressed logic is moved to [`routes.cljs`](../../src/leiningen/new/re_frame/src/cljs/routes.cljs). The two `+routes`-specific `views*.cljs` files have the require (probably from copy-paste), but don't use them at all. I removed the requires from these two instances, created new projects with the affected combinations, built, and tested.

> Also, I see you linked for emacs editor integration

That was present before, for the `+cider` optional profile. This commit edited that line by adding a template tag for clj-kondo, which would follow on the next line if present.

There are no mentions of clj-kondo editor integrations in the README, but I think it's a great idea. I can add it to the Getting started section, where I added the step, "Install clj-kondo (linter)".
